2022/Q4/artifacts/PoS-pseudocode/PoS-model.md

```diff
@@ -292,10 +302,14 @@ func unbond(validator_address, delegator_address, total_amount)
         //(remove it, create a new one with double the amount, and add it).
         validators[validator_address].set_unbonds[cur_epoch+pipeline_length] = {UnbondRecord{amount: amount_unbonded, start: start}} \union validators[validator_address].set_unbonds[cur_epoch+pipeline_length]
         remain -= amount_unbonded
-      update_total_deltas(validator_address, pipeline_length, -1*amount_after_slashing)
+      
+      var pipeline_stake = read_epoched_field(validators[validator_address].total_deltas, cur_epoch + pipeline_length, 0)
+      var token_change = min{ amount_after_slashing, pipeline_stake }
+      update_total_deltas(validator_address, pipeline_length, -1*token_change)
```

Comment:
It looks to me that it would be enough to modify `compute_amount_after_slashing` to always return 0 or greater. If it returns 0, it means that the unbond is fully slashed. I believe the above lines are then not needed.